api/src/services/opportunities_v1/search_opportunities.py

```diff
@@ -182,15 +182,16 @@ def _get_search_request(params: SearchOpportunityParams, aggregation: bool = Tru
 
     # Sorting
     builder.sort_by(_get_sort_by(params.pagination))
-
     # Query
     if params.query:
         filter_rule = FILTER_RULE_MAPPING.get(params.experimental.scoring_rule, DEFAULT)
         builder.simple_query(params.query, filter_rule, params.query_operator)
 
     # Filter Prefix
-    if params.top_level_agency:
-        _add_top_level_agency_prefix(builder, params.top_level_agency, params.filters)
+    if params.filters and params.filters.top_level_agency:
+        _add_top_level_agency_prefix(builder, params.filters)
+        # remove top level agency from filter
+        params.filters.top_level_agency = None
```

Comment:
Could we move all of this inside the `_add_top_level_agency_prefix` function? Make this just: `_add_top_level_agency_prefix(builder, params.filters)` and in that function return if the values are None. That way the logic of how this works isn't split across multiple places, and is contained in one place.